components/style/selector_impl.rs

```diff
@@ -81,11 +79,12 @@ impl SelectorImpl for ServoSelectorImpl {
 
     fn parse_pseudo_element(_context: &ParserContext,
                             name: &str) -> Result<PseudoElement, ()> {
-        use self::PseudoElement::*;
-        match_ignore_ascii_case! { name,
-            "before" => Ok(Before),
-            "after" => Ok(After),
-            _ => Err(())
+        // This prevents an extra allocation in the common case
+        // in which the name is already lowercase.
+        if name.chars().all(|c| c.is_lowercase()) {
+            Ok(PseudoElement::from(name))
+        } else {
+            Ok(PseudoElement::from(&*name.to_ascii_lowercase()))
```

Comment:
@bholley: Please look at the commit message, do you thing we could do well accepting only names that start with a dash, and it that case assume they are already lowercase?